src/math/trig.c

```diff
@@ -12,46 +12,46 @@
 #define _USE_MATH_DEFINES
 #include <math.h>
 #include <stdint.h>
-#define CORDICSINE_TABLE_SIZE	31
-/* Compute fixed point cordicsine with table lookup and interpolation
- *	The cordic sine algorithm converges, when the angle is in the range
+#define CORDIC_SIN_COS_TABLE_SIZE	16
+#define CORDIC_SINE_TABLE_SIZE 31
+/* Compute fixed point cordic sine cos with table lookup and interpolation
+ *	The cordic sine cos algorithm converges, when the angle is in the range
  *	[-pi/2, pi/2).If an angle is outside of this range, then a multiple of
  *	pi/2 is added or subtracted from the angle until it is within the range
  *	[-pi/2,pi/2).Start with the angle in the range [-2*pi, 2*pi) and output
  *	has range in [-1.0 to 1.0]
- *	+------------------+-----------------+--------+--------+
- *	| thRadFxp	   | cdcsinth	     |thRadFxp|cdcsinth|
- *	+----+-----+-------+----+----+-------+--------+--------+
- *	|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
- *	+----+-----+-------+----+----+-------+--------+--------+
- *	| 32 | 28  |  1	   | 32	| 31 |	 1   | 4.28   | 1.31   |
- *	+------------------+-----------------+--------+--------+
+ *	+------------------+-----------------+--------+------------+
+ *	| thRadFxp	   | cdctsincosth    |thRadFxp|cdctsincosth|
+ *	+----+-----+-------+----+----+-------+--------+------------+
+ *	|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat    |
+ *	+----+-----+-------+----+----+-------+--------+------------+
+ *	| 32 | 28  |  1	   | 32	| 31 |	 1   | 4.28   | 1.31       |
+ *	+------------------+-----------------+--------+------------+
  */
 
-int32_t sin_fixed(int32_t th_rad_fxp)
+int32_t sin_cos_fixed(int32_t th_rad_fxp, uint32_t flags)
 {
-	/*cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 31
-	 *CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:31-1))))
+	/*cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 16
+	 *CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:16-1))))
 	 *Inverse CORDIC Gain,inverse_cordic_gain = 1 / cordic_gain
 	 */
-	static const int32_t cordicsine_lookup[CORDICSINE_TABLE_SIZE] = { 843314857, 497837829,
```

Comment:
if i understand correctly the only difference is the LUT, if that is the case we should consolidate these functions and have the table change behind a flag.